Framework/Algorithms/src/ConvertUnits.cpp

```diff
@@ -688,25 +682,19 @@ API::MatrixWorkspace_sptr ConvertUnits::removeUnphysicalBins(
                                                  maxBins - 1);
     // Next, loop again copying in the correct range for each spectrum
     for (int64_t j = 0; j < int64_t(numSpec); ++j) {
-      const MantidVec &X = workspace->readX(j);
-      const MantidVec &Y = workspace->readY(j);
-      const MantidVec &E = workspace->readE(j);
-      MantidVec &Xnew = result->dataX(j);
-      MantidVec &Ynew = result->dataY(j);
-      MantidVec &Enew = result->dataE(j);
       int k;
       for (k = 0; k < lastBins[j] - 1; ++k) {
-        Xnew[k] = X[k];
-        Ynew[k] = Y[k];
-        Enew[k] = E[k];
+        result->mutableX(j)[k] = workspace->x(j)[k];
+        result->mutableY(j)[k] = workspace->y(j)[k];
+        result->mutableE(j)[k] = workspace->e(j)[k];
```

Comment:
Calling `mutable?()` in a tight loop should be avoided (expensive `cow_ptr` access).